src/SnackSession.js

```diff
@@ -292,6 +300,94 @@ export default class SnackSession {
     return url;
   };
 
+  /**
+   * Generates a default app.json template for the snack.
+   * @returns {Object} the app.json template
+   * @function
+   */
+  generateAppJson = (): Object => {
+    const appJsonTemplate = {
+      expo: {
+        description: this.description,
+        slug: 'snack-' + this.snackId,
+        privacy: 'unlisted',
+        sdkVersion: this.sdkVersion,
+        version: '1.0.0',
+        orientation: 'portrait',
+        primaryColor: '#cccccc',
+        icon: 'https://d1wp6m56sqw74a.cloudfront.net/~assets/c9aa1be8a6a6fe81e20c3ac4106a2ebc',
+        loading: {
+          icon: 'https://d1wp6m56sqw74a.cloudfront.net/~assets/c9aa1be8a6a6fe81e20c3ac4106a2ebc',
+          hideExponentText: false,
+        },
+        packagerOpts: {
+          assetExts: ['ttf', 'mp4', 'otf'],
+        },
+        ios: {
+          supportsTablet: true,
+        },
+      },
+    };
+    return appJsonTemplate;
+  };
+
+  /**
+   * Builds an apk from the snack and returns a url to download the apk.
+   * @param 
+   * @returns {Promise.<void>} A promise that contains the url when fulfilled.
+   * @function
+   */
+  getADKUrlAsync = async (
+    appJson : Object,
+    opts: {
+      current?: boolean,
+      mode?: string,
+      platform?: string,
+      expIds?: Array<string>,
+      type?: string,
+      releaseChannel?: string,
+      bundleIdentifier?: string,
+      publicUrl?: string,
+      sdkVersion?: string,
+    } = {}
+  ): Promise<string> => {
+    opts.sdkVersion = this.sdkVersion;
+    opts.platform = 'android';
+    // call out to build api here with url
+    const { id: buildId } = await buildAdkAsync(appJson, opts);
+    const completedJob = await this.wait(buildId, appJson, {});
+    const artifactUrl = completedJob.artifactId
+        ? `https://expo.io/artifacts/${completedJob.artifactId}`
+        : completedJob.artifacts.url;
+
+    return artifactUrl;
+  };
+
+  async wait(buildId, appJson, { timeout = 1200, interval = 60 } = {}) {
+    let time = new Date().getTime();
+    await sleep(secondsToMilliseconds(interval));
+    const endTime = time + secondsToMilliseconds(timeout);
+    while (time <= endTime) {
+      const res = await buildAdkAsync(appJson, {
+        current: false,
+        mode: 'status'
+      });
+      const job = fp.compose(
```

Comment:
Pulling in fp stuff feels like overkill for this.  At least within js, I think the tradeoffs in clarity aren't worth it until you start wanting to have functions that return functions

What happens if buildId is something falsey?
Is it possible to have multiple jobs that make it past the filter condition?